src/db/pool.cr

```diff
@@ -47,6 +60,56 @@ module DB
       @initial_pool_size.times { build_resource }
     end
 
+    def reap_connection!(connection : T, delay : Float64)
+      return unless connection.responds_to?(:check)
```

Comment:
`Connection` implements `#check`, so this guard seems to be unnecessary? However, we could consider to remove the stub implementation and let it be implemented only when it actually does something.